src/Pretzel.Logic/Commands/BaseParameters.cs

```diff
@@ -18,23 +21,30 @@ public sealed class BaseParameters
 
         public bool Safe { get; private set; }
 
-        [Export("SourcePath")]
```

Comment:
I'd prefer to avoid modification on public api.
MEF2 doesn't handle the export of string?

Reply:
There is probably a more elegant way 🤷‍♂️

Reply:
If I am following correctly, it is caused by `Configuration` which involves `Path`?
If that's the case, I don't see a better way so good to me.

Reply:
No the reason is that i see no way of registering the instance of `BaseParameters` in the container that is created in `Program.cs`. That's why i added the second ctor that pulls out the program arguments from `Environment.GetCommandLineArgs().Skip(1).ToArray()`.

I just checked for exporting strings & named string exports and it seams to work. So that should work and I could remove the SourcePathProvider.

Reply:
Removed the `SourcePathProvider`.

Reply:
I have just one idea to avoid the `Environment.GetCommandLineArgs().Skip(1).ToArray()`:

- Don't use `SatisfyImports` on Program, use `GetExport` to resolve `Program`
- Now Compose and all Part discovery needs to be static
- Save the base Parameters in `Main` in an static variable
- Export BaseParameters from Program's instance and get the value in the ctor.

That way only a single instance of BaseParameters would be created

Reply:
Ok, sorry I was focused on the `Path`, I understand now (at least I think I do).
About `BaseParameters`, it is added as Part only to export the path and filesystem, is it necessary to export the entire class in order to export some properties?

Reply:
No it's not required (I was really struggeling with the NullReferenceException if the export does not work, so I tried any thought it's required). Fixed.

Reply:
The importing constructor shouldn't be needed anymore since only the Parse method is used in Program.

Reply:
That's what I was talking about before. Now there are 2 instances. One created by Parse one by MEF. If I restructure the bootstrapping I can workout a way that the SourcePath get's exported by Program. But for now the ImportingCtor need's either to stay, or I do the refactoring.

I've updated a version of Program.cs that avoids the circle/recursion problem.

Reply:
Good to me, thanks!